### PR TITLE
validate chronology ; improve some views and specs

### PR DESCRIPTION
**Pull request by david135**
Addresses #140, #142, #133, and #144.  Factored out of the work I'd begun on tasks.

The most interesting piece is in `config/initializers/validators.rb`, where the validator `validates_chronology` is defined, to be available in all of our models.  It is used as follows:
`validates_chronology :start_time, :end_time`

I've used it in models that seem to need it, and confirmed that they passes existing tests for chronology, and added tests to models that hadn't tested this.

`item.rb` has a couple of fields that may or may not need this validation.  They are there, but commented, and tests have not been added.

I've also fixed datepicker/datetimepicker's functionality in a number of forms (the problem manifested as corruption of a field's contents when editing a form with a saved date/datetime value) .

I've fixed a few cosmetic issues I've discovered.

### CODE REVIEW

File: app/models/item.rb [hunk 1]
```diff
@@ -8,6 +8,9 @@ class Item < ActiveRecord::Base
                   :department_id, :resource_type_id, :item_type_id,
                   :unique_ids_attributes
 
+  # validates_chronology :purchase_date, :sell_date     # ? - if so, needs a test
+  # validates_chronology :grantstart, :grantexpiration  # ? - if so, needs a test
```
Comment (kgf) [inline app/models/item.rb:12]: Grants will be a separate model, so the validations will go there

File: app/models/timecard.rb [hunk 2]
```diff
@@ -10,27 +10,11 @@ class Timecard < ActiveRecord::Base
...
-      errors.add(:intended_end_time, "must be after the start, unless you are the Doctor")
-    end
-  end
-
-  def actual_end_date_cannot_be_before_actual_start
-    if ((!actual_end_time.blank?) and (!actual_start_time.blank?)) and actual_end_time < actual_start_time
-      errors.add :actual_end_time, "must be after the start, unless you are the Doctor"
-    end
-  end
+  validates_presence_of :person_id, :event_id
```
Comment (kgf) [inline app/models/timecard.rb:13]: A future PR will need to remove the link to events. A Timecard represents a person working for a period of time, which could include multiple events.

File: app/views/availabilities/_form.html.erb [hunk 3]
```diff
@@ -3,14 +3,11 @@
...
-                :collection => (Availability::STATUS - ["Cancelled"]) %>
+    <%= f.input :status, :as => :select, :collection => (Availability::STATUS - ["Cancelled"]) %>
     <%= f.input :description %>
 
-    <%= f.input :start_time, :as => :string,
-                :input_html => { :class => 'datetimepicker', :size => 11 }  %>
-    <%= f.input :end_time, :as => :string,
-                :input_html => { :class => 'datetimepicker', :size => 11 }  %>
+    <%= f.input :start_time, as: :string, input_html: { class: 'datetimepicker', size: 11, value: format_datetime_value(f.object.start_time) } %>
+    <%= f.input :end_time, as: :string, input_html: { class: 'datetimepicker', size: 11, value: format_datetime_value(f.object.end_time) } %>
```
Comment (kgf) [inline app/views/availabilities/_form.html.erb]: These are very long lines, which do not fit into the default github UI. Please break them across two lines, similar to the originals. On the plus side, yours are more accurate !

File: app/views/inspections/_form.html.erb [hunk 4]
```diff
@@ -2,8 +2,7 @@
   <%= f.error_notification %>
 
   <div class="form-inputs">
-    <%= f.input :inspection_date, as: :string, input_html: { class: 'datetimepicker', size: 11,
-                                                             value: format_datetime_value(f.object.inspection_date) } %>
+    <%= f.input :inspection_date, as: :string, input_html: { class: 'datetimepicker', size: 11, value: format_datetime_value(f.object.inspection_date) } %>
```
Comment (kgf) [inline app/views/inspections/_form.html.erb]: Not even sure what changed here, other than all on one line.

File: spec/features/inspection_display_spec.rb [hunk 5]
```diff
@@ -13,8 +13,8 @@
     click_on 'Sign in'
   end
   describe "when not logged in" do
-    let(:an_item)  { FactoryGirl.create(:item) }
-    let(:an_inspection) { FactoryGirl.create(:inspection, item: an_item) }
```
Comment (kgf) [inline spec/features/inspection_display_spec.rb:17]: Why are you removing FactoryGirl ?

File: spec/models/cert_spec.rb [hunk 6]
```diff
@@ -4,11 +4,17 @@
...
-  it 'sets the issued date to today' do
+  it 'sets the issued date to today' do # This is set in the factory
     expect(subject.issued_date).to be == Date.today
   end
 
   it 'sets the expiration date properly' do
     expect(subject.expiration_date).to be == Date.today + evoc_course.term.to_i.months
   end
+
+  it "requires expiration_date to be after issued_date" do #chronology
```
Comment (kgf) [inline spec/models/cert_spec.rb:15]: I like this, new tests !

**Comment by kgf** [comment]
Thanks for the new tests !

File: spec/features/inspection_display_spec.rb (hunk 5 above)
Comment (david135) [reply spec/features/inspection_display_spec.rb:17]: I'd realized that they were systematically removed in b975d6f2ffb2ff273b94c0f5221bca28747307fd . If that was the standard, I'd erred in putting them in myself.

**Pull request merged**